lib/routes/defaults.js

```diff
@@ -3,60 +3,72 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 var path = require('path')
-var fs = require('fs')
-var util = require('util')
-var child_process = require('child_process')
+var cp = require('child_process')
 
 var version = require('../../package.json').version
 var commitHash
+var sourceRepo
 
-module.exports = function (log, P, db, error) {
+// Production and stage provide './config/version.json'. Try to load this at
+// startup; punt on failure. For dev environments, we'll get this from `git`
+// for dev environments.
+(function requireVersionInfo() {
+  try {
+    var versionJson = path.join(__dirname, '..', '..', 'config', 'version.json')
+    var info = require(versionJson)
+    commitHash = info.version.hash
+    sourceRepo = info.version.source
+  } catch (e) {
+    /* ignore */
+  }
+})()
 
-  var routes = [
-    {
-      method: 'GET',
-      path: '/',
-      handler: function index(request, reply) {
-        log.begin('Defaults.root', request)
 
-        function sendReply() {
-          reply(
-            {
-              version: version,
-              commit: commitHash
-            }
-          )
-        }
+module.exports = function (log, P, db, error) {
 
-        // if we already have the commitHash, send the reply and return
-        if (commitHash) {
-          return sendReply()
+  function versionHandler(request, reply) {
+    log.begin('Defaults.root', request)
+
+    function sendReply() {
+      reply(
+        {
+          source: sourceRepo,
+          version: version,
+          commit: commitHash
         }
+      )
+    }
 
-        // Note: we figure out the Git hash in the following order:
-        //
-        // (1) read config/version.json if exists (ie. staging, production)
-        // (2) figure it out from git (either regular '.git', or '/home/app/git' for AwsBox)
+    // if we already have the commitHash, send the reply and return
+    if (commitHash) {
+      return sendReply()
+    }
 
-        // (1) read config/version.json if exists (ie. staging, production)
-        var configFile = path.join(__dirname, '..', '..', 'config', 'version.json')
-        if ( fs.existsSync(configFile) ) {
-          commitHash = require(configFile).version.hash
-          return sendReply()
+    cp.exec('git rev-parse HEAD', function(err, stdout1) {
+      if (err) {
+        throw err
+      }
+      cp.exec('git config --get remote.origin.url', function(err, stdout2) {
+        if (err) {
+          throw err
         }
```

Comment:
Hmm. I need to change the above. If there is no remote named 'origin', this will throw. Better to just default to say that sourceRepo is 'undetermined' or some string like that.